app/controllers/projects_controller.rb

```diff
@@ -107,6 +107,16 @@ def sidebar
     }
   end
 
+  def project_filter
+    projects = Project.readable_by_user(current_user)
+                      .search(current_user, false, params[:query], 1, current_team)
+                      .pluck(:id, :name)
+
+    return render plain: [].to_json if projects.blank?
```

Comment:
return 404

Reply:
I think it will be better to return and empty array in this case because that's simply the result of the search.
Same logic for seaching experiments and tasks.